### PR TITLE
refactor webviews

### PR DESCRIPTION
**Pull request by hlshen**

### CODE REVIEW

File: firebase-vscode/webpack.common.js [hunk 1]
```diff
@@ -219,11 +219,11 @@ const extensionConfig = {
   },
 };
 
-function makeWebConfig(entryName) {
+function makeWebConfig(entryName, entryPath = "") {
   return {
     name: entryName,
     mode: "none", // this leaves the source code as close as possible to the original (when packaging we set this to 'production')
-    entry: `./webviews/${entryName}.entry.tsx`,
+    entry: `./webviews/${entryPath}${entryName}.entry.tsx`,
```
Comment (joehan) [inline firebase-vscode/webpack.common.js:226]: Right now, this is fragile to mismatched slashes (ie 'myEntryPath/' works, but 'myEntryPath' does not). You probably want to use path.join to construct this..

File: firebase-vscode/webpack.common.js [hunk 2]
```diff
@@ -309,4 +309,9 @@ module.exports = [
     .filter((filename) => filename.match(/\.entry\.tsx/))
     .map((filename) => filename.replace(/\.entry\.tsx/, ""))
     .map((name) => makeWebConfig(name)),
+  ...fs
```
Comment (joehan) [inline firebase-vscode/webpack.common.js:312]: Nit - this is a lot of logic to inline. For clarity, consider making these 2 long chained calls into variables ie:
```
const webConfigs =   ...fs
    .readdirSync("webviews")
    .filter((filename) => filename.match(/\.entry\.tsx/))
    .map((filename) => filename.replace(/\.entry\.tsx/, ""))
    .map((name) => makeWebConfig(name));
const dataconnectWebConfigs = <blah>

module.exports = [ extensionConfig, ...webConfigs, ...dataconnectWebConfigs ];

**Review by joehan — APPROVED** [review]
LGTM after at least the path fix.

File: firebase-vscode/webpack.common.js (hunk 1 above)
Comment (hlshen) [reply firebase-vscode/webpack.common.js]: done

File: firebase-vscode/webpack.common.js (hunk 2 above)
Comment (hlshen) [reply firebase-vscode/webpack.common.js]: done

**Pull request merged**